src/rpc/core_rpc_server.cpp

```diff
@@ -1069,6 +1069,45 @@ namespace cryptonote
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::on_get_public_nodes(const COMMAND_RPC_GET_PUBLIC_NODES::request& req, COMMAND_RPC_GET_PUBLIC_NODES::response& res, const connection_context *ctx)
+  {
+    PERF_TIMER(on_get_public_nodes);
+
+    COMMAND_RPC_GET_PEER_LIST::response peer_list_res;
+    const bool success = on_get_peer_list(COMMAND_RPC_GET_PEER_LIST::request(), peer_list_res, ctx);
+    res.status = peer_list_res.status;
+    if (!success)
+    {      
+      return false;
+    }
+    if (res.status != CORE_RPC_STATUS_OK)
+    {
+      return true;
+    }
+
+    const auto collect = [](const std::vector<peer> &peer_list, std::vector<public_node> &public_nodes)
+    {
+      for (const auto &entry : peer_list)
```

Comment:
`public_nodes.reserve(peer_list.size())`

Reply:
Not the case, peer_list.size() ≫ public_nodes.size()

Reply:
I don't see why that matters. The `reserve` call will remove allocations in this loop (to one before the loop) and a decent (but admittedly indeterminate) chunk of of the list is going to be copied anyway. And the `std::vector` is nearly immediately deallocated - once the message is serialized and transmitted.

Reply:
I prefer doing a few re-allocations here over allocating almost completely unused big chunk of memory.